File: 231.power_of_two.c
#include <stdio.h>
#include <stdbool.h>

bool isPowerOfTwo(int n)
{
    if (n <= 0)
        return false;

    return (n & (n - 1)) == 0;
}

int main()

{
    printf("%d", isPowerOfTwo(9));
    return 0;
}

File: 231. Power of Two.py
# class Solution:
#     def isPowerOfTwo(self, n):
#         if n == 1:
#             return True
#         while n > 1:
#             if n == 2:
#                 return True
#             elif n % 2 == 0:
#                 n = n / 2
#             else:
#                 return False


class Solution:
    def isPowerOfTwo(self, n):
        if n <= 0:
            return False

        return n & (n - 1) == 0


newIsPowerOfTwo = Solution()
print(newIsPowerOfTwo.isPowerOfTwo(16))
